lib/locations/reader.rb

```diff
@@ -0,0 +1,45 @@
+require 'open-uri'
+
+module Locations
+  class Reader
+    attr_accessor :path, :redis
+
+    def initialize(path, redis = Redis.new(uri: redis_uri))
+      self.path = path
+      self.redis = redis
+    end
+
+    def call
+      if json_from_url?
+        cached_json
+      else
+        read_json(path)
+      end
+    end
+
+    private
+
+    def json_from_url?
+      path.to_s =~ /^http/
+    end
+
+    def cached_json
+      json = redis.get('locations')
+      unless json
+        json = read_json(path)
+        redis.setex('locations', 60 * 60, json)
```

Comment:
Should we be determining the cache duration or observing the headers sent by the HTTP service?